server/src/main/java/io/crate/analyze/SwapTableAnalyzer.java

```diff
@@ -68,10 +71,26 @@ public AnalyzedSwapTable analyze(SwapTable<Expression> swapTable,
         }
         SearchPath searchPath = txnCtx.sessionSettings().searchPath();
         Role user = txnCtx.sessionSettings().sessionUser();
-        return new AnalyzedSwapTable(
-            schemas.findRelation(swapTable.source(), Operation.ALTER, user, searchPath),
-            schemas.findRelation(swapTable.target(), Operation.ALTER, user, searchPath),
-            dropSource
-        );
+
+        var source = schemas.findRelation(swapTable.source(), Operation.ALTER_TABLE_RENAME, user, searchPath);
+        if (!(source instanceof DocTableInfo sourceRelation)) {
+            throw new OperationOnInaccessibleRelationException(
+                source.ident(),
+                String.format(
+                    Locale.ENGLISH, "The relation \"%s\" doesn't support or allow %s operations",
+                    source.ident().name(), Operation.ALTER_TABLE_RENAME
```

Comment:
seeing this again, since the user has called `SWAP`, maybe it's confusing to mention `ALTER RENAME` in the error message, shouldn't we just hardcode `SWAP` here?